relay-server/src/services/project_upstream.rs

```diff
@@ -115,12 +119,35 @@ impl ProjectStateChannel {
     }
 
     pub fn send(self, state: ProjectFetchState) {
+        for channel in self.other {
+            channel.send(state.clone());
+        }
         self.channel.send(state)
     }
 
     pub fn expired(&self) -> bool {
         Instant::now() > self.deadline
     }
+
+    pub fn merge(&mut self, channel: ProjectStateChannel) {
+        let ProjectStateChannel {
+            channel,
+            other,
+            deadline,
+            no_cache,
+            attempts,
+            errors,
+            pending,
+        } = channel;
+
+        self.other.push(channel);
+        self.other.extend(other);
+        self.deadline = self.deadline.max(deadline);
```

Comment:
Should this be the minimum? In theory the deadline could get bumped indefinitely here, which maybe isn't a bad thing?

Reply:
I think max is correct, matches the previous behavior more closely.

Reply:
I think `min` matches the previous behaviour more closely since we'd replace the newer entry with the older entry and the deadline is only computed once.